tests/rust/src/lib.rs

```diff
@@ -39,6 +39,12 @@ pub unsafe fn create_tmp_dir(dir_fd: wasi::Fd, name: &str) -> wasi::Fd {
             | wasi::RIGHTS_FD_READDIR
             | wasi::RIGHTS_FD_FILESTAT_GET
             | wasi::RIGHTS_FD_SEEK
+            | wasi::RIGHTS_PATH_LINK_SOURCE
```

Comment:
Given that those tests were passing on  wasmtime, I wonder if wasmtime correctly implemented file right system. Looks like files created in parent directory inherit `base` rights, instead of inheriting `inheriting` rights. @abrown is that something you could confirm?

Reply:
Can you point me to which version of Wasmtime is being used to run the tests? I do see recent-ish PRs that may have changed this behavior:
- https://github.com/bytecodealliance/wasmtime/pull/6784
- https://github.com/bytecodealliance/wasmtime/pull/6479

Reply:
This is on 12.0.2

Reply:
I would try the latest version of Wasmtime then and see if that resolves things?

Reply:
I just tried the latest main branch, and tests in its current form (e.g. `path_filestat`) are still passing.

Reply:
> Looks like files created in parent directory inherit base rights, instead of inheriting inheriting rights.

Just want to confirm that's the behaviour I observed.

@abrown if it's helpful, here's a list of tests which should fail (without the changes from this PR) and the required missing rights:

* `path_link.rs`
```
wasi::RIGHTS_PATH_LINK_SOURCE
wasi::RIGHTS_PATH_LINK_TARGET
wasi::RIGHTS_PATH_OPEN
wasi::RIGHTS_PATH_UNLINK_FILE
```
* `path_filestat` and `symlink_filestat.rs`
```
wasi::RIGHTS_PATH_FILESTAT_GET
```